src/Filesystem.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Yiisoft\Files;
+
+use League\Flysystem\Filesystem as LeagueFilesystem;
+use League\Flysystem\FilesystemAdapter;
+use League\Flysystem\PathNormalizer;
+use Yiisoft\Aliases\Aliases;
+
+class Filesystem extends LeagueFilesystem implements FilesystemInterface
+{
+    private Aliases $aliases;
+
+    public function __construct(FilesystemAdapter $adapter, array $aliases = [], array $config = [], PathNormalizer $pathNormalizer = null)
+    {
+        if ($aliases !== []) {
+            $aliases = array_merge(['@root' => ''], $aliases);
+            $aliases['@root'] = '';
+        }
+        $this->aliases = new Aliases($aliases);
```

Comment:
Looks like a hidden dependency. I think it should be removed overall but at least moving to constructor as Aliases instance would be more correct.